features/fixtures/mazerunner/app/src/main/java/com/bugsnag/android/mazerunner/StartupANRBehaviour.kt

```diff
@@ -20,23 +22,26 @@ fun Application.triggerStartupAnrIfRequired() {
         // we have to startup Bugsnag at this point
         Bugsnag.start(this)
 
-        thread {
-            // This is a dirty hack to work around the limitations of our current testing
-            // systems - where external key-events are pushed through our main thread (which we
-            // are pausing to test for ANRs)
+        // wait for Bugsnag's ANR handler to install first
+        Handler(Looper.getMainLooper()).post {
```

Comment:
Loading plugins has required an adjustment to the test scenario to wait until the first looper message has finished executing, as otherwise a startup ANR is missed. This is a bit of a trade off between always catching startup ANRs, and having a strictmode violation show up in the ANR plugin.

It might be possible to suppress the violation within `AnrPlugin`, or simply ignore it when creating a Bugsnag listener.

Reply:
In-person discussion decided that either the plugin should be responsible for loading specific parts on a bg thread, or the plugin should entirely load on the main thread and the violation can be ignored within the penaltyListener